server/plugin.go

```diff
@@ -153,16 +153,74 @@ func (p *Plugin) handleNotification(body io.Reader) {
 		return
 	}
 
+	var attachment model.SlackAttachment
```

Comment:
I'm wondering if a small refactor would benefit this code.  Functionally, I believe you are correct, but think something like the following would help clarify and allow for extending to other notification types.  I think removing this `if/else` block style would be beneficial for readability.

I would create a smaller `sendPostNotification` function containing `post` initialization and the `model.ParseSlackAttachement()` and `CreatePost` calls. 

Something like the following: 
```go
	isRdsEvent, messageNotification := p.isRDSEvent(notification.Message)
	if isRdsEvent {
		p.API.LogDebug("Processing RDS Event")
		p.sendPostNotification(p.createSNSRdsEventAttachment(notification.Subject, messageNotification))
		return
	}

	if isAlarm, messageNotification := p.isCloudWatchAlarm(notification.Message); isAlarm {
		p.API.LogDebug("Processing CloudWatch alarm")
		p.sendPostNotification(p.createSNSMessageNotificationAttachment(notification.Subject, messageNotification))
		return 
	}


func (p *Plugin) sendPostNotification(attachment model.SlackAttachment) {
	post := &model.Post{
		ChannelId: p.ChannelID,
		UserId:    p.BotUserID,
	}
	model.ParseSlackAttachment(post, []*model.SlackAttachment{&attachment})
	if _, appErr := p.API.CreatePost(post); appErr != nil {
		return
	}
}



```

Reply:
Thanks @jfrerich , suggestion is implemented!